setup/command/aws.go

```diff
@@ -0,0 +1,66 @@
+package command
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/defaults"
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/setup/aws"
+	"github.com/urfave/cli"
+)
+
+var awsFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:   "aws-access-key",
+		Usage:  "access key portion of an AWS key",
+		EnvVar: config.AWS_ACCESS_KEY_ID,
+	},
+	cli.StringFlag{
+		Name:   "aws-secret-key",
+		Usage:  "secret key portion on an AWS key",
+		EnvVar: config.AWS_SECRET_ACCESS_KEY,
+	},
+	cli.StringFlag{
+		Name:   "aws-region",
+		Usage:  "AWS region",
+		EnvVar: config.AWS_REGION,
+	},
+}
+
+func (f *CommandFactory) newAWSProviderHelper(c *cli.Context) (*aws.Provider, error) {
+	// use default credentials and region settings
+	config := defaults.Get().Config
+
+	// use static credentials if passed in by the user
+	accessKey := c.String("aws-access-key")
+	secretKey := c.String("aws-secret-key")
+	if accessKey != "" && secretKey != "" {
+		staticCreds := credentials.NewStaticCredentials(accessKey, secretKey, "")
+		config.WithCredentials(staticCreds)
+	} else {
+		logrus.Debugf("aws-access-key or aws-secret-key was not specified. Using default credentials")
```

Comment:
the same defaults used by the `aws` cli - which is `~/.aws/credentials` and the `AWS_ACCESS/SECRET_KEY` environment variables